core/src/cluster_info.rs

```diff
@@ -1499,52 +1504,49 @@ impl ClusterInfo {
         thread_pool: &ThreadPool,
         pulls: &mut Vec<(Pubkey, CrdsFilter, SocketAddr, CrdsValue)>,
     ) {
-        let pull_from_entrypoint = {
-            let mut w_entrypoint = self.entrypoint.write().unwrap();
-            if let Some(ref mut entrypoint) = &mut *w_entrypoint {
+        let entrypoint_id_and_gossip = {
+            let mut entrypoints = self.entrypoints.write().unwrap();
+            if entrypoints.is_empty() {
+                None
+            } else {
+                let i = thread_rng().gen_range(0, entrypoints.len());
```

Comment:
This is the main logic change in `core/src/cluster_info.rs`.   Rather than **the** entrypoint, every time we run though `append_entrypoint_to_pulls()` randomly pick **one** of the entrypoints and check if they need a pull using the same logic as before